tools/fv_diagnostics.F90

```diff
@@ -3012,6 +3014,7 @@ subroutine fv_diag(Atm, zvir, Time, print_freq)
           endif
        endif
 
+       if(id_delp_dycore > 0) used=send_data(id_delp_dycore, Atm(n)%delp(isc:iec,jsc:jec,:), Time)
```

Comment:
@lharris4 - should this be moved inside the #ifdef to not have the same diagnostic output with two different names when the macro GFS_PHYS is not defined?

Reply:
I think you are right, there is no need to carry two definitions in non-`GFS_PHYS` implementations. 

Ideally `delp` or `delp_total` would be the regular total-mass definition (dependent on nwat), `delp_moist` or `delp_dry` defined appropriately, for all configurations, and the NetCDF metadata would make clear which is which. 

`delp` is typically not useful for most end-users, only to those creating diagnostics very closely consistent with the numerics, so this would not be too disruptive.

Reply:
> should this be moved inside the #ifdef to not have the same diagnostic output with two different names when the macro GFS_PHYS is not defined?

@bensonr indeed in the current state I think this makes sense.  

> Ideally `delp` or `delp_total` would be the regular total-mass definition (dependent on nwat), `delp_moist` or `delp_dry` defined appropriately, for all configurations, and the NetCDF metadata would make clear which is which.
> 
> `delp` is typically not useful for most end-users, only to those creating diagnostics very closely consistent with the numerics, so this would not be too disruptive.

@lharris4 do you have a concrete suggestion?  I am open to making a breaking change—the mismatch between the `delp` diagnostic and the `delp` variable in restart files caught me off guard originally.  I'm also fine with changing the name of the diagnostic added here, e.g. `delp_total` is what I named it in FV3GFS, and maybe improving the metadata of the diagnostics for the two versions of the pressure thickness.

Reply:
I went ahead with a conservative approach—renaming `delp_dycore` to `delp_total`, and only exposing it as a diagnostic when compiled using `GFS_PHYS`.  I also revised the description to hopefully be a little more explicit:

```
'FV3 pressure thickness (dry air + all water species)'
```  

I notice the description of `delp` in SHiELD does at least include a reference to `'GFS moist-mass'` if compiled with `GFS_PHYS` ([which is better than FV3GFS](https://github.com/ai2cm/fv3gfs-fortran/blob/e3c522d285b2c9dac925f21e8c3d1f36df9f2dda/FV3/atmos_cubed_sphere/tools/fv_diagnostics.F90#L747-L748)), so I have left that alone.

@lharris4 @bensonr let me know what you think.  Thanks!